rfcs/text/0002-rfc-environment.md

```diff
@@ -47,73 +49,118 @@ Stage 2: Include new or updated yml field definitions for all of the essential f
 
-Note: we may want to improve the configuration of Beats to not require to define a processor to specify `service.environment`, similarly to the existing solution to define `fields.*` fields.
+The configuration could look like:
 
 ```yaml
+# EXPERIMENT - NOT SUPPORTED CONFIGURATION SNIPPET
+heartbeat.monitors:
+- type: http
+  id: www-frontend-monitor
+  name: Website Frontend Monitor
+  service:
+    name: www-frontend
+    environment: production
+  urls: ["https://www.my-ecommerce-company.com/status"]
+  schedule: '@every 10s'
+  check.response.status: 200
+  timeout: 2s
+- type: http
+  id: www-frontend-monitor-staging
+  name: Website Frontend Monitor - Staging
+  service:
+    name: www-frontend
+    environment: staging
+  urls: ["https://www.staging.my-ecommerce-company.com/status"]
+  schedule: '@every 10s'
+  check.response.status: 200
+  timeout: 5s
+``` 
+
+Note the support in Heartbeat of `service.name` is waiting for https://github.com/elastic/beats/pull/20330
+
+### Usage with Filebeat and Metricbeat
+
+Note that all the Beats come with a different example field to define the environment: `fields.env` (see filebeat.yml, metricbeat.yml, auditbeat.yml or heartbeat.yml).
+This `fields.env` field is misaligned with this desire of standardisation because it cannot be standardised in ECS due to the following ECS rules:
+* The namespace `fields.*` is not accepted in ECS, this namespace is dedicated to non standardised fields.
+* ECs don't use abbreviations and `env` is the abbreviation of `environment`.
+
+Note that the `fields.env` field is just an example and it not likely to be used very broadly because for infrastructure elements such as servers, the delineation between environments is often difficult to establish as servers are frequently simulatenaously running production and non production workloads.
+
+We propose to tackle the question of standardisation environment for infrastructure later.
+
+
+We don't need to evolve Filebeat and Metricbeat configurations initially.
+
+Note: we may want to improve the configuration of Filebeat and Metricbeat later down the road to ease the specification `service.environment` for specific files or metrics that can be associated with a specific service/application.
+Today, such configuration is cumbersome requiring to specify a "processor". 
+
+Example of verbose Filebeat configuration today to define `service.name=www-frontend` and `service.environment=production`:
+
+```
 ---
 filebeat.inputs:
   - type: log
     enabled: true
-    json.keys_under_root: true
-    json.overwrite_keys: true
     paths:
-      - /usr/local/var/log/my-shopping-cart/anti-fraud.log
+      - /var/log/www-frontent/www-frontend.log
 processors:
   - add_fields:
+      when:
+        ... todo find the syntax to filter on file path
       target: 'service'
       fields:
-        environment: staging
+        name: www-frontend
+        environment: production
 ```
 
-
-
 ## Source data
```

Comment:
Since this proposal is about correlating events from different sources via the service fields and the proposed `service.environment` field, I think we should use this section to give relevant event snippets that demonstrate this.

Could you grab one or a few sample APM events, and a sample Heartbeat event in JSON format? Then we can adjust their `service` fields to fit with the example you've been building in the "Usage" section.

And finally to demonstrate the correlation with Metricbeat or Filebeat, we could do the same with one of their events and also adjust its `service` fields to demonstrate how each event across the stack could be correlated. If needed I can help you gather a Filebeat example.

Reply:
Problem solved